packages/stripe-integration/src/stripe-ocs/stripe-ocs-payment-strategy.ts

```diff
@@ -139,25 +134,30 @@ export default class StripeOCSPaymentStrategy implements PaymentStrategy {
         gatewayId: string,
         methodId: string,
     ) {
-        const state = await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
-            params: { method: methodId },
-        });
-        const paymentMethod = state.getPaymentMethodOrThrow(methodId);
+        let paymentMethod = this.paymentIntegrationService
+            .getState()
+            .getPaymentMethodOrThrow(methodId);
+
+        if (!paymentMethod?.clientToken) {
+            const state = await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
+                params: { method: methodId },
+            });
+
+            paymentMethod = state.getPaymentMethodOrThrow(methodId);
+        }
 
         if (!isStripePaymentMethodLike(paymentMethod)) {
             throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
-        const {
-            clientToken,
-            initializationData: { stripePublishableKey, stripeConnectedAccount, shopperLanguage },
-        } = paymentMethod;
+        const { clientToken, initializationData } = paymentMethod;
+        const { shopperLanguage, customerSessionToken } = initializationData;
```

Comment:
Is this BC customer session token?

Reply:
No, it is Stripe customer session.
[https://docs.stripe.com/api/checkout/sessions](https://docs.stripe.com/api/checkout/sessions)